pkg/scalers/metrics_api_scaler.go

```diff
@@ -63,6 +119,56 @@ func metricsAPIMetadata(resolvedEnv, metadata, authParams map[string]string) (*m
 		return nil, fmt.Errorf("no valueLocation given in metadata")
 	}
 
+	if val, ok := authParams["authMode"]; ok {
```

Comment:
How about using negation in this if and then returning instantly?
```
if val, ok := authParams["authMode"]; !ok {
    return here
}
```
In this way we will limit the indentation

Reply:
oh,  good catch. looks much cleaner now.